qiskit_machine_learning/neural_networks/estimator_qnn.py

```diff
@@ -0,0 +1,249 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Estimator quantum neural network class"""
+
+from __future__ import annotations
+
+import logging
+from copy import copy
+from typing import Sequence
+
+import numpy as np
+from qiskit.algorithms.gradients import (
+    BaseEstimatorGradient,
+    EstimatorGradientResult,
+    ParamShiftEstimatorGradient,
+)
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator, Estimator, EstimatorResult
+from qiskit.quantum_info import SparsePauliOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from qiskit_machine_learning.exceptions import QiskitMachineLearningError
+
+from .neural_network import NeuralNetwork
+
+logger = logging.getLogger(__name__)
+
+
+class EstimatorQNN(NeuralNetwork):
+    """A Neural Network implementation based on the Estimator primitive.
+
+    The ``EstimatorQNN`` is a neural network that takes in a parametrized quantum circuit
+    with the combined network's feature map (input parameters) and ansatz (weight parameters)
+    and outputs its measurements for the forward and backward passes.
```

Comment:
I think that this definition could be updated to sth like:

```suggestion
    The ``EstimatorQNN`` is a neural network that takes in and a parametrized quantum circuit
    with the combined network's feature map (input parameters) and ansatz (weight parameters), as well as 
    an observable, and outputs their expectation value computations for the forward and backward passes.
```

Reply:
@woodsp-ibm are you fine with this proposal?

Reply:
It takes a single circuit parameterized circuit right where you designate parameters for input and/or weights. Will a newcomer be confused by seeing terminology like feature map and anstaz here? Sure you can concatentate one of the library feature map circuits with one its ansatz circuits (as TwoLocal did before) but is this a good basic description or would it be more an example. Maybe the part given over to input params we want to refer to it as a feature map but I think only this above text here does that right, nowhere else in the qnn - same for ansatz.

Reply:
I'm not sure I got the point. I updated the description slightly. Feel free to suggest changes if you think it's necessary.

Reply:
My point was more that when you come to QNNs this just takes a circuit however you make it. The description seemed to lean to heavily into feature map/ansatz as terms to describe its operation which this does not have (unlike TwoLocal where it has like named parameters). The text is much better now in that it makes the feature map + anstaz an example of what can be done so it seems you got the basic point I was trying to make, I think.

Maybe we could even go a bit further and have a small code example in the class docstring that builds an instance with a feature map and ansatz combined circuit, that follows on from that text, which I think may helpful for someone coming to ML. Sure we will have tutorials elsewhere that should cover this, but something very simple here in the API ref to show this.

Reply:
Added a code snippet.